paimon-core/src/main/java/org/apache/paimon/catalog/CachingCatalog.java

```diff
@@ -309,32 +292,35 @@ private class TableInvalidatingRemovalListener implements RemovalListener<Identi
         public void onRemoval(Identifier identifier, Table table, @NonNull RemovalCause cause) {
             LOG.debug("Evicted {} from the table cache ({})", identifier, cause);
             if (RemovalCause.EXPIRED.equals(cause)) {
-                tryInvalidateSysTables(identifier);
+                tryInvalidateAttachedTables(identifier);
             }
         }
     }
 
     @Override
     public void invalidateTable(Identifier identifier) {
         tableCache.invalidate(identifier);
-        tryInvalidateSysTables(identifier);
+        tryInvalidateAttachedTables(identifier);
         if (partitionCache != null) {
             partitionCache.invalidate(identifier);
         }
     }
 
-    private void tryInvalidateSysTables(Identifier identifier) {
-        if (!isSpecifiedSystemTable(identifier)) {
-            tableCache.invalidateAll(allSystemTables(identifier));
-        }
-    }
-
-    private static Iterable<Identifier> allSystemTables(Identifier ident) {
-        List<Identifier> tables = new ArrayList<>();
-        for (String type : SYSTEM_TABLES) {
-            tables.add(Identifier.fromString(ident.getFullName() + SYSTEM_TABLE_SPLITTER + type));
+    private void tryInvalidateAttachedTables(Identifier identifier) {
```

Comment:
I'm wondering if the system table doesn't need to be stored in the cache anymore. Just wrap it when retrieving it

Reply:
I'm ok to cache the system table or not, because identity is our cache key, and the system table is a kind of identity.

But I'm just wondering why do we need to cache the origin table before caching the system table (maybe this step can be skipped)?

Reply:
I checked the previous logic again. Now we reuse the table (in cache) and then wrap it to the system table. In this case, I think we don't need to cache the system table.

Another solution is to only cache the system table (not warp from cache table). 

Which one do you think is more suitable?

Reply:
I think we can just cache the original table.